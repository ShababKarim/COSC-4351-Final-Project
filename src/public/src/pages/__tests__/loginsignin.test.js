import React, { useState } from 'react';
import { render, cleanup } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import LoginSignUpPage from '../loginsignin';

import AuthContext from '../../context';

afterEach(cleanup);

describe('Login-Signin page', () => {
	it('renders without crashing', () => {
		const { result } = renderHook(() =>
			useState({
				_id: 1,
				name: 'test',
				email: 'test@gmail.com'
			})
		);
		const change = newObj => (result.current = newObj);
		const { asFragment } = render(
			<AuthContext.Provider
				value={{ auth: result.current, setAuth: change }}
			>
				<LoginSignUpPage />
			</AuthContext.Provider>
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
