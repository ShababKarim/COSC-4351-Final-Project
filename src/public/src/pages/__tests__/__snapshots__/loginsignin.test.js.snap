// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login-Signin page renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthSm"
  >
    <div
      class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div
          style="position: relative; width: 100px; height: 100px;"
        >
          <img
            alt="cur"
            src="UH_Red.png"
            style="position: absolute; display: block; left: 50px; top: 50px; width: 0px; height: 0px;"
          />
        </div>
      </div>
    </div>
    <div
      class="makeStyles-root-1"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorDefault"
      >
        <div
          aria-label="full width tabs example"
          class="MuiTabs-root"
        >
          <div
            class="MuiTabs-scroller MuiTabs-fixed"
            style="overflow: hidden;"
          >
            <div
              class="MuiTabs-flexContainer"
              role="tablist"
            >
              <button
                aria-controls="full-width-tabpanel-0"
                aria-selected="true"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary makeStyles-tab-2 Mui-selected MuiTab-fullWidth"
                id="full-width-tab-0"
                role="tab"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiTab-wrapper"
                >
                  Employee Login
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                aria-controls="full-width-tabpanel-1"
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary makeStyles-tab-2 MuiTab-fullWidth"
                id="full-width-tab-1"
                role="tab"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiTab-wrapper"
                >
                  Employee Sign Up
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <span
              class="PrivateTabIndicator-root-173 PrivateTabIndicator-colorPrimary-174 MuiTabs-indicator"
              style="left: 0px; width: 0px; background: rgb(211, 47, 47);"
            />
          </div>
        </div>
      </header>
      <div
        style="overflow-x: hidden;"
      >
        <div
          class="react-swipeable-view-container"
          style="flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform;"
        >
          <div
            aria-hidden="false"
            data-swipeable="true"
            style="width: 100%; flex-shrink: 0; overflow: auto;"
          >
            <div
              aria-labelledby="full-width-tab-0"
              class="MuiTypography-root MuiTypography-body1"
              id="full-width-tabpanel-0"
              role="tabpanel"
            >
              <div
                class="MuiBox-root MuiBox-root-208"
              >
                <div
                  class="MuiContainer-root MuiContainer-maxWidthSm"
                >
                  <form
                    class="makeStyles-container-209"
                  >
                    <div>
                      <div
                        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
                      >
                        <label
                          class="MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-required Mui-required"
                          data-shrink="true"
                          for="email"
                          id="email-label"
                        >
                          Email
                          <span
                            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                          >
                             *
                          </span>
                        </label>
                        <div
                          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorSecondary MuiInput-colorSecondary MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                        >
                          <input
                            aria-invalid="false"
                            class="MuiInputBase-input MuiInput-input"
                            id="email"
                            name="email"
                            placeholder="email"
                            required=""
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
                      >
                        <label
                          class="MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-required Mui-required"
                          data-shrink="true"
                          for="password"
                          id="password-label"
                        >
                          Password
                          <span
                            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                          >
                             *
                          </span>
                        </label>
                        <div
                          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorSecondary MuiInput-colorSecondary MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                        >
                          <input
                            aria-invalid="false"
                            autocomplete="current-password"
                            class="MuiInputBase-input MuiInput-input"
                            id="password"
                            name="password"
                            placeholder="password"
                            required=""
                            type="password"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-211 MuiButton-containedPrimary MuiButton-fullWidth"
                      tabindex="0"
                      type="submit"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Login
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
